### ISSUE TITLE
No support for proxy at version 2+

### ISSUE THREAD

**Issue by smalcts** [pnp/pnpjs#1195]
### Category
- [x] Enhancement

### Version

Please specify what version of the library you are using: 2.0.5

### Desired Behavior
In versions 1.3.2+ of the library it was possible to add a proxy setting as described here: https://pnp.github.io/pnpjs/v1/nodejs/docs/proxy/. Now the possibility seems to be gone.

### Observed Behavior
Is it possible to add this option to 2.0.5+?

I've tried to solve this issue for myself with the following code, but it doesn't seem to work (even though the log messages appear).

```javascript
        class SPFetchClientProxied extends SPFetchClient {
            async fetch(url: string, options?: any) {
                const agent = httpProxyAgent();
                if (agent) {
                    console.log('Setting http proxy agent');
                    options.agent = agent;
                }
                return super.fetch(url, options);
            }
        }
```

**Comment by patrick-rodgers** [comment]
Yes, that's a miss on our part. We added it in v1 after we branched for v2 and never moved that over. Sorry about that. We'll get it added for the next release. Thanks for letting us know!

**Comment by smalcts** [comment]
Thank you for the reply. Yes, that would be a nice new feature.

In the meantime, maybe, you know what's wrong with my workaround? From what I see, the request in the end is done using node-fetch and node-fetch should accept agent in its options (and it does if I'm calling node-fetch directly).

**Comment by patrick-rodgers** [comment]
This is fixed in #1200 and will be included in the next version. Thanks again for reporting this.
